File: frontend/src/types/report.ts
import type { PayableEntityType } from './payment';

export interface PayableReportRow {
  entity_id: string;
  entity_name: string;
  entity_type: PayableEntityType;
  subtype?: string | null;
  // month number (1-12) -> total paid that month. JSON object keys arrive as
  // strings, so index with the numeric month (JS coerces to string).
  months: Record<string, number>;
  total: number;
}

export interface PayablesReport {
  year: number;
  entity_type: string;
  rows: PayableReportRow[];
}

export interface AssetRegisterRow {
  id: string;
  name: string;
  asset_type: string;
  status: string;
  acquisition_date: string | null;
  acquisition_cost: number | null;
  current_value: number | null;
  taxes_paid: number;
  premiums_paid: number;
}

export interface AssetRegisterReport {
  rows: AssetRegisterRow[];
}

File: backend/app/schemas/report.py
import uuid
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class PayableReportRow(BaseModel):
    entity_id: uuid.UUID
    entity_name: str
    entity_type: str
    # bill_type / tax_type / insurance_type - lets the client carve out
    # "other payables" and group by kind.
    subtype: str | None = None
    # month number (1-12) -> total amount paid that month
    months: dict[int, Decimal]
    total: Decimal


class PayablesReport(BaseModel):
    year: int
    entity_type: str
    rows: list[PayableReportRow]


class AssetRegisterRow(BaseModel):
    id: uuid.UUID
    name: str
    asset_type: str
    status: str
    acquisition_date: date | None
    acquisition_cost: Decimal | None
    current_value: Decimal | None
    taxes_paid: Decimal
    premiums_paid: Decimal


class AssetRegisterReport(BaseModel):
    rows: list[AssetRegisterRow]
